src/student/schemas/student.schema.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable prettier/prettier */
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { randomBytes, scrypt as _scrypt } from 'crypto';
+import { NextFunction } from 'express';
+import { promisify } from 'util';
+
+@Schema()
+export class Student {
+    
+    @Prop({required: true})
+    name: string;
+
+    @Prop({required: true, unique: true})
+    email: string;
+
+    @Prop({required: true})
+    password: string;
+
+    @Prop({required: true})
+    phone: number;
+
+    @Prop({required: true})
+    sem: number;
+
+    @Prop({required: true})
+    departmentId: string;
+
+    @Prop({required: true})
```

Comment:
set default role as student as you know every user created with this schema is student